File: wakakusa/demo/cards.rb
# A piece of screen with a name is a method that answers an element,
# and one that wraps other elements takes them as arguments.
require "wakakusa"

class Cards
  def initialize
    @a = 0
    @b = 0
  end

  def card(title, *kids)
    column(
      text(title, size: 18.0),
      *kids,
      spacing: 4.0, padding: 8.0,
      border_width: 1.0, border_color: "accent", border_radius: 8.0
    )
  end

  def view
    column(
      card("counters",
           row(text("a: #{@a}"), button("+1") { @a += 1 }, spacing: 6.0),
           row(text("b: #{@b}"), button("+10") { @b += 10 }, spacing: 6.0)),
      text("outside the card", size: 12.0),
      spacing: 10.0,
      padding: 16.0
    )
  end
end

run(Cards.new, title: "cards")

File: gomamochi/demo/cards.go
// A piece of screen with a name is a method that answers an element,
// and one that wraps other elements takes them as arguments.
package main

import (
	"fmt"

	. "github.com/i2y/yokan/gomamochi"
)

type Cards struct {
	a int
	b int
}

func (c *Cards) card(title string, kids ...Element) Element {
	cells := []Element{Text(title).Size(18)}
	cells = append(cells, kids...)
	return Column(cells...).Spacing(4).Padding(8).
		BorderWidth(1).BorderColor("accent").BorderRadius(8)
}

func (c *Cards) View() Element {
	return Column(
		c.card("counters",
			Row(Text(fmt.Sprintf("a: %d", c.a)), Button("+1").OnClick(func() { c.a += 1 })).Spacing(6),
			Row(Text(fmt.Sprintf("b: %d", c.b)), Button("+10").OnClick(func() { c.b += 10 })).Spacing(6)),
		Text("outside the card").Size(12),
	).Spacing(10).Padding(16)
}

func main() {
	Run(&Cards{}, Title("cards"))
}
